backends/lean/Base/Primitives/Scalar.lean

```diff
@@ -1419,6 +1419,19 @@ instance (ty: ScalarTy) : LinearOrder (Scalar ty) where
     right; exact (Scalar.le_equiv _ _).2 H
   decidableLE := ScalarDecidableLE ty
 
+-- Coercion theorems
+-- This is helpful whenever you want to "push" casts to the innermost nodes
+-- and make the cast normalization happen more magically.
+
+@[simp, norm_cast]
+theorem coe_max {ty: ScalarTy} (a b: Scalar ty): Max.max a b = (Max.max a b: ℤ) := by
```

Comment:
What does this theorem do?
On the left a and b are scalars, and on the right they are coerced to integers first, correct?
If yes, it would make more sense to put the coercion on a and b (rather than `Max.max a b`).

Reply:
`Max.max (coe a) (coe b) = coe (Max.max a b)`, this is what this theorem does.

Reply:
OK, I will put explicitly the coercion in the arguments as well.

Reply:
@sonmarcho done.